src/components/Popups/ReplyField/ReplyField.tsx

```diff
@@ -69,6 +78,28 @@ export default class ReplyField extends React.Component<Props, State> {
         this.saveCursorPosition();
     }
 
+    getItemList = (mentionString: string): Collaborator[] => {
```

Comment:
Should this be `getCollaborators`? It also seems like it could live in another component within PopupList that accepts a `filter` string and an `options` array as props. We should really be striving for smaller components, when reuse is a possibility.

Reply:
> Should this be `getCollaborators`? It also seems like it could live in another component within PopupList that accepts a `filter` string and an `options` array as props. We should really be striving for smaller components, when reuse is a possibility.

I tried doing that but then realized that I need to know the filtered list length in the `handleKeyDown` handler which must live in this time.

Reply:
The filtered list is derived by filtering `props.collaborators` using `getActiveMentionForEditorState (this.state.editorState)`. We should avoid storing derived state when the constituent data is already available in state and/or props.

Reply:
> The filtered list is derived by filtering `props.collaborators` using `getActiveMentionForEditorState (this.state.editorState)`. We should avoid storing derived state when the constituent data is already available in state and/or props.

That makes sense, but I feel that it is wasteful to perform all of these computations in handleKeyDown every time just for length.

Reply:
Wasteful in what way? Is there a measurable performance impact? If not, we should prefer to follow the common convention of reducing state as much as possible.